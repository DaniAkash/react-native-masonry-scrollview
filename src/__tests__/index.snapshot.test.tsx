import React from "react";
import renderer from "react-test-renderer";
import RNMasonryScrollView from "../index";
import { Text } from "react-native";

const column1Text = "Text 1";
const column2Text = "Text 2";
const column3Text = "Text 3";

const VMasonryComponentTwoColumns = () => {
  return (
    <RNMasonryScrollView>
      {[<Text key={1}>{column1Text}</Text>, <Text key={2}>{column2Text}</Text>]}
    </RNMasonryScrollView>
  );
};

it("Renders Vertical Masonry with 2 columns", () => {
  const tree = renderer.create(<VMasonryComponentTwoColumns />).toJSON();
  expect(tree).toMatchSnapshot();
});

const VMasonryComponentThreeColumns = () => {
  return (
    <RNMasonryScrollView>
      {[
        <Text key={1}>{column1Text}</Text>,
        <Text key={2}>{column2Text}</Text>,
        <Text key={3}>{column3Text}</Text>
      ]}
    </RNMasonryScrollView>
  );
};

it("Renders Vertical Masonry with 3 columns", () => {
  const tree = renderer.create(<VMasonryComponentThreeColumns />).toJSON();
  expect(tree).toMatchSnapshot();
});

const HMasonryComponentTwoColumns = () => {
  return (
    <RNMasonryScrollView>
      {[<Text key={1}>{column1Text}</Text>, <Text key={2}>{column2Text}</Text>]}
    </RNMasonryScrollView>
  );
};

it("Renders Horizontal Masonry with 2 columns", () => {
  const tree = renderer.create(<HMasonryComponentTwoColumns />).toJSON();
  expect(tree).toMatchSnapshot();
});

const HMasonryComponentThreeColumns = () => {
  return (
    <RNMasonryScrollView>
      {[
        <Text key={1}>{column1Text}</Text>,
        <Text key={2}>{column2Text}</Text>,
        <Text key={3}>{column3Text}</Text>
      ]}
    </RNMasonryScrollView>
  );
};

it("Renders Horizontal Masonry with 3 columns", () => {
  const tree = renderer.create(<HMasonryComponentThreeColumns />).toJSON();
  expect(tree).toMatchSnapshot();
});
