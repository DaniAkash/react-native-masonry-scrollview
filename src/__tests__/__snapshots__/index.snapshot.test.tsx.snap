// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Horizontal Masonry with 2 columns 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 2
    </Text>
  </View>
</ScrollView>
`;

exports[`Renders Horizontal Masonry with 2 columns 2`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 1
    </Text>
    <Text>
      Text 2
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 2
    </Text>
  </View>
</ScrollView>
`;

exports[`Renders Vertical Masonry with 2 columns 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 1
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 2
    </Text>
  </View>
</ScrollView>
`;

exports[`Renders Vertical Masonry with 3 columns 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 1
    </Text>
    <Text>
      Text 2
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        null,
        null,
      ]
    }
  >
    <Text>
      Text 2
    </Text>
  </View>
</ScrollView>
`;
